### PR TITLE
Loop alignment: Handle blocks added in loop as part of split edges of LSRA

### PR DESCRIPTION
**Pull request by kunalspathak**
If there are new blocks added by LSRA that modifies the flow of blocks that are in loop, then make sure that we do not align such
loops if they intersect with the last aligned loop.

Fixes some manifestation of https://github.com/dotnet/runtime/issues/47094.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/emit.cpp [hunk 1]
```diff
@@ -4820,58 +4820,112 @@ unsigned emitter::getLoopSize(insGroup* igLoopHeader, unsigned maxLoopSize DEBUG
...
+//    If there is an intersection of current loop with last loop that is already marked as align,
+//    then *do not align* one of the loop that completely encloses the other one. Or if they both intersect,
+//    then *do not align* either of them because since the flow is complicated enough that aligning one of them
+//    will not improve the performance.
 //
 void emitter::emitSetLoopBackEdge(BasicBlock* loopTopBlock)
 {
-    insGroup* dstIG = (insGroup*)loopTopBlock->bbEmitCookie;
+    insGroup* dstIG              = (insGroup*)loopTopBlock->bbEmitCookie;
+    bool      noAlignCurrentLoop = false;
```
Comment (BruceForstall) [inline src/coreclr/jit/emit.cpp:4834]: Can you reverse the sense of these, i.e., `alignCurrentLoop` and `alignLastLoop`? It's very confusing sometimes when "no" or "not" is part of a variable name.

**Review by BruceForstall — APPROVED** [review]
Any asm diffs?

File: src/coreclr/jit/emit.cpp (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/jit/emit.cpp:4834]: Sure.

**Comment by kunalspathak** [comment]
> Any asm diffs?

No asm diffs. The scenario mainly come in jitstressregs mode where we end up adding new blocks for resolution.

**Comment by kunalspathak** [comment]
Formatting errors will be addressed in https://github.com/dotnet/runtime/pull/55320

**Comment by kunalspathak** [comment]
One of the failure is https://github.com/dotnet/runtime/issues/55315

**Pull request merged**